addons/workspace-dots/userscript.js

```diff
@@ -0,0 +1,81 @@
+export default async function ({ addon, console }) {
+  const blockly = await addon.tab.traps.getBlockly();
+
+  const oldFunction = blockly.getMainWorkspace().grid_.update;
+
+  function updateGrid() {
+    let themeSetting = addon.settings.get("theme");
+    let spacingSetting = addon.settings.get("useSpacing");
+    let spacingAmount = addon.settings.get("spacing");
+
+    blockly.getMainWorkspace().grid_.update = function (scale) {
+      console.log(themeSetting, spacingSetting, spacingAmount);
+
+      this.scale_ = scale;
+
+      let spacing = this.spacing_;
+      if (spacingSetting) spacing = spacingAmount;
+
+      // MSIE freaks if it sees a 0x0 pattern, so set empty patterns to 100x100.
+      var safeSpacing = spacing * scale || 100;
+
+      this.gridPattern_.setAttribute("width", safeSpacing);
+      this.gridPattern_.setAttribute("height", safeSpacing);
+
+      var half = Math.floor(spacing / 2) + 0.5;
+      var start = half - this.length_ / 2;
+      var end = half + this.length_ / 2;
+
+      half *= scale;
+      start *= scale;
+      end *= scale;
```

Comment:
this code is (mostly) the same as on https://github.com/scratchfoundation/scratch-blocks/blob/develop/core/grid.js#L136. why not just run `oldFunction.call(this, spacing)`?

Reply:
It's a stupid reason, i recently got terrified of .call after so many issues with what this object to pass through it with the disable auto nesting addon, but I will change it. Let me dm you tho

Reply:
Just realise i literally cant because i need to do calculations inside of that function, no way around it, so it has to be modified fully and not just called with scripts before or after

Reply:
I think something like this would work:
```js
blockly.getMainWorkspace().grid_.update = function (scale) {
  const oldSpacing = this.spacing_;
  if (spacingSetting & !addon.self.disabled) this.spacing_ = spacingAmount;
  oldFunction.call(this, scale);

  let strokeWidthX;
  let strokeWidthY;
  // ...
  this.line1_.setAttribute("stroke-width", strokeWidthY);
  this.line2_.setAttribute("stroke-width", strokeWidthX);

  this.spacing_ = oldSpacing;
}

Reply:
Is there any real reason? its already a small enough function

Reply:
The original function might change in a Scratch update or be overridden by another addon in a future SA version. If you decide to keep the copied code anyway, there should be a comment indicating where it's from.